moto/mediapackage/models.py

```diff
@@ -0,0 +1,202 @@
+        self.mss_package = kwargs.get("mss_package")
+        self.origination = kwargs.get("origination")
+        self.startover_window_seconds = kwargs.get("startover_window_seconds")
+        self.tags = kwargs.get("tags")
+        self.time_delay_seconds = kwargs.get("time_delay_seconds")
+        self.url = kwargs.get("url")
+        self.whitelist = kwargs.get("whitelist")
+
+    def to_dict(self):
+        data = {
+            "arn": self.arn,
+            "authorization": self.authorization,
+            "channelId": self.channel_id,
+            "cmafPackage": self.cmaf_package,
+            "dashPackage": self.dash_package,
+            "description": self.description,
+            "hlsPackage": self.hls_package,
+            "id": self.id,
+            "manifestName": self.manifest_name,
+            "mssPackage": self.mss_package,
+            "origination": self.origination,
+            "startoverWindowSeconds": self.startover_window_seconds,
+            "tags": self.tags,
+            "timeDelaySeconds": self.time_delay_seconds,
+            "url": self.url,
+            "whitelist": self.whitelist,
+        }
+        return data
+
+
+class MediaPackageBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(MediaPackageBackend, self).__init__()
+        self.region_name = region_name
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+        self._channels = OrderedDict()
+        self._origin_endpoints = OrderedDict()
+
+    def create_channel(self, description, id, tags):
+        arn = "arn:aws:mediapackage:channel:{}".format(id)
+        channel = Channel(
+            arn=arn,
+            description=description,
+            egress_access_logs={},
+            hls_ingest={},
+            channel_id=id,
+            ingress_access_logs={},
+            tags=tags,
+        )
+        self._channels[id] = channel
+        return channel
+
+    def list_channels(self):
+        channels = list(self._channels.values())
+        response_channels = [c.to_dict() for c in channels]
+        return response_channels
+
+    def describe_channel(self, id):
+        channel = self._channels[id]
+        return channel.to_dict()
+
+    def delete_channel(self, id):
+        channel = self._channels[id]
+        del self._channels[id]
+        return channel.to_dict()
+
+    def create_origin_endpoint(
+        self,
+        authorization,
+        channel_id,
+        cmaf_package,
+        dash_package,
+        description,
+        hls_package,
+        id,
+        manifest_name,
+        mss_package,
+        origination,
+        startover_window_seconds,
+        tags,
+        time_delay_seconds,
+        whitelist,
+    ):
+        arn = "arn:aws:mediapackage:origin_endpoint:{}".format(id)
+        url = "https://origin-endpoint.mediapackage.eu_west_1.amazonaws.com/{}".format(
```

Comment:
I'm assuming the region in this URL will be dependent on the specified region?
Would be good to have an assertion in one of the tests as well, to verify the URL is returned

Reply:
It's not dependent on region yet - it should be
```
"https://origin-endpoint.mediapackage.{}.amazonaws.com/{}".format(self.region_name, id)
```

Reply:
@bblommers About that - for some reason `self.region_name` ends up being None when we run our tests? Do you know why that may be? 

Also, just as a clarification, that URL is a bit arbitrary and not really used by anything, we just needed a string that kind of looks like a URL.

Reply:
In models.py, you have to pass the region when creating the Backend per region:
`mediapackage_backends[region] = MediaPackageBackend()` -> `mediapackage_backends[region] = MediaPackageBackend(region)`

Reply:
Thank you, I hadn't realised this. I have made the change now.